### PR TITLE
Adjusted replicate node confirmation form text, fixes #4346

### LINKED ISSUE #4346: Replicate button confirmation dialog may need to be adjusted for our use case
image

We were exploring the "Replicate" functionality this afternoon. It's amazing. One minor thing that we noticed is that there's some inconsistency in the terms used in the default confirmation dialog before replicating a page. 

For instance, there's several references to "labels" when currently we're only replicating pages, so "title" may be appropriate here instead.

Also, the phrase "This action cannot be undone" is a little scary and might be a bit off-putting or potentially confusing if the user thinks it's referring to the label field above it. Also, it's technically not true, since you can "undo" a page replication by deleting the replicated page.

### PR DESCRIPTION
**Pull request by quamsta**
Trying to make the replicate page confirmation dialog a bit more friendly and more tailored to our current use case for the module: replicating pages. 

Changes:

- The title "Are you sure you want to replicate _node_ entity id [id]?" now reads "_Replicate Page_ [Page Name]," which matches the "_Edit Page_ [Page Name]" action title.
- The field label "New label (English)" now says "Replicated page's title"
- The description "This text will be used as the label of the new entity being created, in English." now says "This text will be used as the title of the newly replicated page."
- Removed the potentially confusing "This action cannot be undone." message 

Before:
Screen Shot 2021-10-22 at 8 54 48 AM

After:
Screen Shot 2021-10-22 at 8 51 28 AM

# How to test

Replicate a page to see these adjustments (after clearing the cache).

### CODE REVIEW

**Comment by joewhitsitt** [comment]
So, replicate is also enabled for articles. It is also capable of being extended to other content types by developers. Can we look into ways to generalize when this fires and then gets the bundle name to use in the language?

Also, I think "This action cannot be undone" should be left in but can be modified to be more clear. This statement is common around Drupal for things (deleting content, etc.)

File: docroot/modules/custom/uiowa_core/uiowa_core.module
Comment (pyrello) [inline docroot/modules/custom/uiowa_core/uiowa_core.module]: You can use double quotes to avoid needing to escape the single quote here.

Comment (pyrello) [reply docroot/modules/custom/uiowa_core/uiowa_core.module]: FWIW, this is the only time I will suggest using double quotes :)

File: docroot/modules/custom/uiowa_core/uiowa_core.module
Comment (pyrello) [inline docroot/modules/custom/uiowa_core/uiowa_core.module]: Per @joewhitsitt's comment, get the entity bundle to generalize "Page" to be whatever bundle type is being replicated. Since technically, this could apply to non-node entities, it might be good to add a check if the entity type is node and then set this title based on the bundle.

**Review by pyrello — CHANGES_REQUESTED** [review]
See individual comments.

**Comment by quamsta** [comment]
> So, replicate is also enabled for articles. It is also capable of being extended to other content types by developers. Can we look into ways to generalize when this fires and then gets the bundle name to use in the language?

Oh! Yes, I'll see if I can find a way to genericize it. Good call

> Also, I think "This action cannot be undone" should be left in but can be modified to be more clear. This statement is common around Drupal for things (deleting content, etc.)

Up for any ideas, but I feel "This action cannot be undone" is pretty unclear in this case. Makes sense for deleting items or other potentially destructive actions, but it might be a little harsh for something like copying an item.

File: docroot/modules/custom/uiowa_core/uiowa_core.module
Comment (joewhitsitt) [reply docroot/modules/custom/uiowa_core/uiowa_core.module]: would have to change how you get to the right form to alter though. currently `node_page_replicate_form` maybe there is another identifier that you can latch on to

**Comment by joewhitsitt** [comment]
>Up for any ideas, but I feel "This action cannot be undone" is pretty unclear in this case. Makes sense for deleting items or other potentially destructive actions, but it might be a little harsh for something like copying an item.

Yeah, that is fair. It wasn't too long ago that this action was causing database corruption. Glad Cory found a fix and we are where it is today, but I am crossing my fingers each time that button is pushed. I guess the publisher doesn't need to know all that and "for your developer's sake, please use sparingly" doesn't quite roll off the tongue. I will back down on this change.

**Comment by quamsta** [comment]
> > Up for any ideas, but I feel "This action cannot be undone" is pretty unclear in this case. Makes sense for deleting items or other potentially destructive actions, but it might be a little harsh for something like copying an item.
> 
> Yeah, that is fair. It wasn't too long ago that this action was causing database corruption. Glad Cory found a fix and we are where it is today, but I am crossing my fingers each time that button is pushed. I guess the publisher doesn't need to know all that and "for your developer's sake, please use sparingly" doesn't quite roll off the tongue. I will back down on this change.

Yeah, totally understand the reasoning behind wanting to keep a message like that with the difficulties the module presented, but from a user perspective the "cannot be undone language" may present the question of "should I not be doing this?" and I also had a user ask "Is it referring to the title field not being able to be changed after the fact?" since the message was showing up directly under the title field.

**Comment by quamsta** [comment]
Also @joewhitsitt, I just committed a change that should allow for any allowed replicated content type to be referred to in this dialog (instead of just pages). I did a hacky "ucfirst()" to get the content type label to be uppercase in the title attribute, but if you know a better way to do that, please help haha.

**Comment by joewhitsitt** [comment]
More processing but I think something like this would work and handle a world of multi-worded names in the future (e.g. Areas of Study):

```php
$bundle_label = \Drupal::entityTypeManager()
  ->getStorage('node_type')
  ->load($node->bundle())
  ->label();
```

>Since technically, this could apply to non-node entities, it might be good to add a check if the entity type is node and then set this title based on the bundle.

A lot of our code so far is only focused for "content" based on the replicate settings, so let's hold off of on future-proofing for all entities.

File: docroot/modules/custom/uiowa_core/uiowa_core.module
Comment (quamsta) [reply docroot/modules/custom/uiowa_core/uiowa_core.module]: Ended up using the `if ($route_name == 'entity.node.replicate') {` statement that's already being used instead of `node_page_replicate_form`

**Comment by quamsta** [comment]
> More processing but I think something like this would work and handle a world of multi-worded names in the future (e.g. Areas of Study):
> 
> ```
> $bundle_label = \Drupal::entityTypeManager()
>   ->getStorage('node_type')
>   ->load($node->bundle())
>   ->label();
> ```
> 
>  > Since technically, this could apply to non-node entities, it might be good to add a check if the entity type is node and then set this title based on the bundle.
> 
> A lot of our code so far is only focused for "content" based on the replicate settings, so let's hold off of on future-proofing for all entities.

Okay, this is now using the bundle's label() instead of node_type. This does create a minor grammatical issue where the label is now capitalized when it maybe shouldn't be (e.g.,the  "Replicated Article's title" field label should probably be "Replicated article's title," since I've seen "article" and "page" used in lowercase before, but it's probably not a huge deal?

**Comment by joewhitsitt** [comment]
pulling down to test

**Comment by joewhitsitt** [comment]
I guess I misunderstood what you went with ucfirst() originally when providing you a way to get the human readable label.

Hope you don't mind, but I tweaked this a bit to get the best of both worlds. I still think the human readable label should be used but that it should be lower case. I think you protect yourself from apostrophe errors in the future "content" can be squeezed in. (e.g. senior thesis content's title).

Screen Shot 2021-10-22 at 4 03 30 PM

Screen Shot 2021-10-22 at 4 31 24 PM

I am going to approve, but if you disagree with my change, we can revert and go a different way.

**Review by joewhitsitt — APPROVED** [review]

**Comment by quamsta** [comment]
> I guess I misunderstood what you went with ucfirst() originally when providing you a way to get the human readable label.
> 
> Hope you don't mind, but I tweaked this a bit to get the best of both worlds. I still think the human readable label should be used but that it should be lower case. I think you protect yourself from apostrophe errors in the future "content" can be squeezed in. (e.g. senior thesis content's title).
> 
> Screen Shot 2021-10-22 at 4 03 30 PM
> 
> Screen Shot 2021-10-22 at 4 31 24 PM
> 
> I am going to approve, but if you disagree with my change, we can revert and go a different way.

This is great! Thanks a ton Joe! Have a good weekend.

**Review by pyrello — APPROVED** [review]
Approving based on @joewhitsitt's approval.

**Pull request merged**
